R/bcdc_options.R

```diff
@@ -101,8 +100,31 @@ bcdc_single_download_limit <- function() {
     ))
 
     res <- cc$parse("UTF-8")
-    doc <- xml2::read_xml(res)
+    xml2::read_xml(res)
+
+  } else {
+    message("No access to internet")
+    invisible(FALSE)
+  }
+}
+
+bcdc_get_capabilities <- function() {
+  doc <- ._bcdataenv_$get_capabilities_xml
+
+  features <- xml2::xml_child(doc, 4)
+  feature_list <- xml2::as_list(features)
+  purrr::map_dfr(feature_list, ~ {
+    list(whse_name = .x$Name[[1]] %||% NA_character_,
+         title = .x$Title[[1]] %||% NA_character_,
+         cat_url = attr(.x$MetadataURL, "href") %||% NA_character_)
+  })
+}
+
+
+bcdc_single_download_limit <- function() {
+  doc <- ._bcdataenv_$get_capabilities_xml
 
+  if (inherits(doc, "xml_document")) {
     constraints <- xml2::xml_find_all(doc, ".//ows:Constraint")
     count_defaults <- constraints[which(xml2::xml_attrs(constraints) %in% "CountDefault")]
     xml2::xml_double(count_defaults)
```

Comment:
I know this code already existed, but I discovered this to select nodes/attribute combos with xpath:

```suggestion
    count_defaults <- xml2::xml_find_first(doc, ".//ows:Constraint[@name='CountDefault']")
    xml2::xml_integer(count_defaults)
```

https://www.w3schools.com/xml/xpath_syntax.asp

Reply:
🤯